flang/include/flang/Optimizer/Dialect/CUF/CUFOps.td

```diff
@@ -154,13 +154,15 @@ def cuf_DataTransferOp : cuf_Op<"data_transfer", []> {
     ```
   }];
 
-  let arguments = (ins Arg<AnyReferenceLike, "", [MemWrite]>:$src,
-                       Arg<AnyReferenceLike, "", [MemRead]>:$dst,
+  let arguments = (ins Arg<AnyRefOrBoxType, "", [MemWrite]>:$src,
```

Comment:
Can you please clarify in the comment above what happens when the operands are box references? I assume it is following the assignment semantics, so both the LHS data and the descriptor are updated.

Reply:
Yes it should follow the assignment semantic. I added a comment in that sense.